### PR TITLE
Improve scaling of animator and adding aspect keyword to lineanimator

### LINKED ISSUE #2777: LineAnimator display of small y-axis values
The aspect ratio of `LineAnimator` appears to be determined through absolute scaling of x and y values. For example if y ~ 10^-9 and x ~ 1, the y-axis is not visible. We need something like an `aspect` kwarg as in `plt.imshow` so that the length of each axis is determined by the plotting area and not the data range. This issue probably applies to the other animated classes.

### PR DESCRIPTION
**Pull request by Akram9**
### Description

Improve scaling of animator and adding aspect keyword to lineanimator, reference to  #2777

### CODE REVIEW

File: sunpy/visualization/animator/base.py [hunk 1]
```diff
@@ -257,8 +257,8 @@ def _make_axes_grid(self):
             vert = [ysize, pad_size] * self.num_sliders + \
                    [large_pad_size, large_pad_size, Size.AxesY(self.axes)]
         else:
-            vert = [ysize, pad_size] * self.num_sliders + \
-                   [large_pad_size, Size.AxesY(self.axes)]
+            vert = [ysize, large_pad_size] * self.num_sliders + \
+                   [Size.AxesY(self.axes)]
```
Comment (nabobalis) [inline sunpy/visualization/animator/base.py]: Deleting one of the pad_sizes is ok?

Comment (Akram9) [reply sunpy/visualization/animator/base.py]: Frankly, I need to test it a bit more.

Comment (Akram9) [reply sunpy/visualization/animator/base.py]: Well this does solve most of the problem, bu tvery satisfactorily. Also, it's specific to the example in the docs.

**Comment by nabobalis** [comment]
Is this finished?

File: sunpy/visualization/animator/base.py (hunk 1 above)
Comment (nabobalis) [reply sunpy/visualization/animator/base.py]: How well does this fix the issue?

Comment (Akram9) [reply sunpy/visualization/animator/base.py]: Changing y-axis values no longer affects the plot size as earlier but changing the window size does affect the whole plot. Everything goes out of ratio in y-axis again and depending on how it's done, the slider comes on top of the plot.

**Comment by Akram9** [comment]
Not yet.

**Comment by Akram9** [comment]
I would actually like someone to check out the changes.

**Comment by Akram9** [comment]
To run -
```
import numpy as np
import matplotlib.pyplot as plt

from sunpy.visualization.animator import LineAnimator

data_shape0 = (10, 20)
data0 = np.random.rand(*data_shape0)

plot_axis0 = 1
slider_axis0 = 0

xdata = np.tile(np.linspace(0, 100, data_shape0[plot_axis0]+1), (data_shape0[slider_axis0], 1))
ani = LineAnimator(data0, plot_axis_index=plot_axis0, axis_ranges=[None, xdata])

plt.show()
```

**Comment by nabobalis** [comment]
I had a quick try with that code and it does scale a bit better than normal.

I am not sure if it solves the original issue without checking.

**Comment by Akram9** [comment]
Thanks. Also try to give `aspect` a try by - `ani = LineAnimator(data0, plot_axis_index=plot_axis0, axis_ranges=[None, xdata], aspect=float)`. It behaves pretty wierd at times.

**Comment by nabobalis** [comment]
Yeah, I'm unsure on the aspect keyword.

**Comment by Akram9** [comment]
I'll leave the aspect for now and see if I can better the scaling.

**Comment by Akram9** [comment]
Please check the last change. I think this solves all problems here. 
The aspect feature still remains glitchy though.

**Comment by nabobalis** [comment]
I think the fix helps the scaling issues with our current plots and adding aspect to lineanimator helps. But I am not sure if it solves the original issue. But I am happy to merge this in regardless.

**Comment by nabobalis** [comment]
Thanks @Akram9!

**Comment by nabobalis** [comment]
Having checked the code given in the original issue, this does in fact close #2777.

**Pull request merged**